Tests/OpenAPIURLSessionTests/URLSessionTransportTests.swift

```diff
@@ -27,41 +27,54 @@ import Foundation
 @preconcurrency import class FoundationNetworking.URLSessionConfiguration
 #endif
 @testable import OpenAPIURLSession
+import HTTPTypes
 
 class URLSessionTransportTests: XCTestCase {
 
-    func testRequestConversion() throws {
-        let request = OpenAPIRuntime.Request(
-            path: "/hello%20world/Maria",
-            query: "greeting=Howdy",
+    func testRequestConversion() async throws {
+        let request = HTTPRequest(
             method: .post,
+            scheme: nil,
+            authority: nil,
+            path: "/hello%20world/Maria?greeting=Howdy",
             headerFields: [
-                .init(name: "X-Mumble", value: "mumble")
-            ],
-            body: Data("👋".utf8)
+                .init("x-mumble2")!: "mumble"
+            ]
+        )
+        let body: HTTPBody = "👋"
+        let urlRequest = try await URLRequest(
+            request,
+            body: body,
+            baseURL: URL(string: "http://example.com/api")!
         )
-        let urlRequest = try URLRequest(request, baseURL: URL(string: "http://example.com/api")!)
         XCTAssertEqual(urlRequest.url, URL(string: "http://example.com/api/hello%20world/Maria?greeting=Howdy"))
         XCTAssertEqual(urlRequest.httpMethod, "POST")
-        XCTAssertEqual(urlRequest.allHTTPHeaderFields, ["X-Mumble": "mumble"])
```

Comment:
This was fun. Turns out URLRequest just changes the casing of the stored header, so we can't simply compare `allHTTPHeaderFields` with the expected ones, as it internally capitalizes things: https://github.com/apple/swift-corelibs-foundation/blob/8a9b69b5041b5069360239cd18304fd8e2eb91d9/Sources/FoundationNetworking/NSURLRequest.swift#L479

Reply:
Can't we move the `.canonicalName` to the test when doing the conversion as a normalization before the equality check?

Reply:
The problem here is the opposite - URLRequest takes an all-lowercase name, and makes some of the characters uppercase, for some reason (linked above). We just need to do the equality using one of the methods that's case-insensitive (as that's the contract for HTTP headers), we shouldn't really care about URLRequest's internal header name representation (even though we used to in this test).